packages/wrangler/src/deploy/deploy.ts

```diff
@@ -168,6 +168,53 @@ function errIsStartupErr(err: unknown): err is ParseError & { code: 10021 } {
 	return false;
 }
 
+export const validateRoutes = (
+	routes: Route[],
+	hasExperimentalAssets: boolean
+) => {
+	const invalidRoutes: { [route: string]: string[] } = {};
+	for (const route of routes) {
+		if (typeof route !== "string" && route.custom_domain) {
```

Comment:
do these apply for Workers with assets as well?

Reply:
the custom domain ones apply for everything, including workers + assets